### ISSUE TITLE
EIO Reporting Workflow

### ISSUE THREAD

**Issue by patriciarussellCAS** [bcgov/cas-reporting#483]
#### Description:
Industry users who register as an Electricity Import Operation, or EIO will have a simplified report that is different than the regular Annual Report. They will NOT have a compliance obligation. 

[Wireframe workflow here](https://www.figma.com/design/LsDmLDCdnJqI8UahyqvOD8/BCIERS-Designs?node-id=18885-5088&t=9EePik5cYAopZX3Y-4)

There is a separate ticket for EIO's final review page: #399 
#### Acceptance Criteria:

Given I have selected EIO as my registration purpose
When I fill out my annual report 
Then I have the following forms: 
 1. Review Operation Information form (simplified version with no activities or products or BOROID)
 2. Person Responsible form (no changes required)
 3. Electricity Import Data **NEW FORM - should match this [wireframe](https://www.figma.com/design/LsDmLDCdnJqI8UahyqvOD8/BCIERS-Designs?node-id=18474-12679&t=yvmr4T1Q1uXJPl8M-4)
 4. Final Review Page - see ticket #399 
 5. Verification Page - We should display this page to all EIOs but do not make it mandatory
 6. Attachments Page - Do not need to make the verification statement field mandatory for EIOs.

Given I am a reporting, EIO, or filing a simple report,
Then this is the new [sign off page](https://www.figma.com/design/LsDmLDCdnJqI8UahyqvOD8/BCIERS-Designs?node-id=18472-10022&t=kZrmBRORBQb9GskT-4) that I see

#### Dev Checklist
- [x] In `bciers/apps/reporting/src/app/components/taskList/types.ts` add HeaderStep ` ElectricityImportData = "Electricity Import Data",`
- [x] In `bciers/apps/reporting/src/app/components/taskList/types.ts` add ReportingPage ` ElectricityImportData = "Electricity Import Data",`
- [x]  Create EIO report flow `bciers/apps/reporting/src/app/components/taskList/reportingFlows/eioReportFlow.ts`
- [x] In `bciers/apps/reporting/src/app/components/taskList/reportingFlows.ts` add `eioReportFlow.ts`
- [x] In `bciers/apps/reporting/src/app/components/taskList/reportingFlows.ts` add condition based on EIO
- [x] Create EIO task list `bciers/apps/reporting/src/app/components/taskList/taskListPages/2_eioInformation.ts`
- [x] In `bciers/apps/reporting/src/app/components/taskList/taskListPages/pageFactories.ts` add `2_eioInformation.ts`
- [x] Update `bciers/apps/reporting/src/app/utils/getReportNeedsVerification.ts` for EIO by-pass
- [x] Update `bciers/apps/reporting/src/middlewares/withRuleHasReportRouteAccess.ts` to use `getReportNeedsVerification.ts` 
- [x] Add "stub" schemas `bciers/apps/reporting/src/data/jsonSchema/eio/eio.ts`
- [x] Add "stub" `bciers/apps/reporting/src/app/components/eio/ElectricityInformationPage.tsx`
- [x] Add "stub"  `bciers/apps/reporting/src/app/components/eio/ElectricityInformationForm.tsx`
- [x] Add app routes to `ElectricityInformationPage`
- [x] Add routing rules in `bciers/apps/reporting/src/middlewares/withRuleHasReportRouteAccess.ts` for EIO
- [x]  Create RJSF form for `Electricity Import Data`
- [x] Create UI Schema for `Electricity Import Data`
- [x] Modify Header Element for Step- Report Information, use multistepFormWithTaskList Component in ElectricityImportForm, modify eioFlow, add isElectricityImport to TaskListPageFactoryContext
- [x] Implement Sign Off Page for EIO and Supplementary EIO
- [x] Update Review Operation Page for EIO.
- [x] Implement model for `Electricity Import Data`
- [x] Implement service for `Electricity Import Data`
- [x] implement API for `Electricity Import Data`
- [ ] Meets the DOD

#### Definition of Ready (Note: If any of these points are not applicable, mark N/A)

- [ ] User story is included
- [ ] User role and type are identified
- [ ] Acceptance criteria are included
- [ ] Wireframes are included (if required)
- [ ] Design / Solution is accepted by Product Owner
- [ ] Dependencies are identified (technical, business, regulatory/policy)
- [ ] Story has been estimated (under 13 pts)

#### Definition of Done (Note: If any of these points are not applicable, mark N/A)

- [ ] Acceptance criteria are tested by the CI pipeline
- [ ] UI meets accessibility requirements
- [ ] Configuration changes are documented, documentation and designs are updated
- [ ] Passes code peer-review
- [ ] Passes QA of Acceptance Criteria with verification in Dev and Test
- [ ] Ticket is ready to be merged to main branch
- [ ] Can be demoed in Sprint Review
- [ ] Bugs or future work cards are identified and created
- [ ] Reviewed and approved by Product Owner

#### Notes:

-

**Comment by patriciarussellCAS** [comment]
NOTE: Before we complete the work on EIO revisit whether or not we should import EIO data from SWRS so we can connect our records to the historical records via metabase.

**Comment by fviduya** [comment]
I have tested this in the DEV environment, [here is the test document](https://buttoninc-my.sharepoint.com/:w:/g/personal/francesca_button_is/EYQR3VTeDTNJpuPgd2_nV-kB2MaAYi5ES20rog46ReDTVg?e=YG00YI).

I can verify that when operation's registration purpose is Electricity Import Operation:
- [x] The Review Operation Information page does not include activities, products, and BORO ID
- [x] There are no changes to the Person Responsible page
- [x]  After clicking `Save & Continue` from the Person Responsible page, the page routes to Electricity Import Data page similar to the [wireframe](https://www.figma.com/design/LsDmLDCdnJqI8UahyqvOD8/BCIERS-Designs?node-id=18474-12679&t=yvmr4T1Q1uXJPl8M-4) - **NOTE: The fields here accept negative numbers, accepts letters by default, and the error message by default is "Enter letters only" spoke with @ayeshmcg and additional changes for this page will be tackled in #399 **
- [x] The Verification page is displayed by default
- [x] The Verification page is not mandatory - Will be tackled in #743
- [x] The Attachments page is displayed by default
- [x] The Verification Statement is not mandatory - Will be tackled in #743
- [x] The Sign-off page for the supplementary report version has additional statement - @patriciarussellCAS, the additional statement says 'new version of this annual report', wanted to just verify if that is correct 

Hi @patriciarussellCAS, the wireframe in the link is for the supplementary report, is this the [correct wireframe](https://www.figma.com/design/LsDmLDCdnJqI8UahyqvOD8/BCIERS-Designs?node-id=18472-10022&t=1wLvIOgYLHqM1p8s-0)? 

The linked wireframe in the description also do not have Verification page in the task list and has also incorrect headers. cc: @zoeyli-46 

Moving to Pending PO for review of test and notes. Thank you!

**Comment by patriciarussellCAS** [comment]
@ayeshmcg - just to confirm. when I'm reviewing this in Dev, I can see the final review page but it's missing the Electricity Import Data form information, but this will be added with ticket #399 correct? 
@fviduya - I think we need a separate priority ticket for the verification page being all mandatory fields when it should not be. I don't know how that will be fixed by #734 (same with the attachments page) - that seems like the opposite issue. 
We need EIO's to be able to click through Verification and attachments without requiring anything. 
Since these were part of the AC for this ticket I'm not sure whether to Accept this ticket or not.

**Comment by ayeshmcg** [comment]
@patriciarussellCAS 
Yes, you're right — the Electricity Import Data form information will be added with ticket #399, which covers the EIO Final Review page.

Regarding the Verification and Attachments pages: since this ticket was already quite large and covered a lot of scope, we decided to handle those issues separately in ticket #743. That ticket ensures that in the EIO workflow, users will be able to click through the Verification and Attachments pages without being blocked by mandatory field validations.

**Comment by fviduya** [comment]
Sorry, @patriciarussellCAS, it's supposed to be #743 not 734!! 🤦

**Comment by patriciarussellCAS** [comment]
lol, thanks @fviduya that makes WAY more sense!

**Comment by patriciarussellCAS** [comment]
Moving to accepted = noting the outstanding items are being address in #743 and #399

**Comment by TammieV-BCG** [comment]
UAT Task

Log in with BCeID
Given that you are filling out an annual report for an EIO
Observe that you see a simplified report where;

- [x] Review Operation Information form has no activities, products or BOROID

- [x] There is a Person Responsible form 

- [x] Electricity Import Data form matches   this wireframe

- [x] There is a Final Review Page in read only mode  

- [x] The Verification Page is displayed but not mandatory - 

- [x] Verification statement field are not mandatory on **the Attachments Page**

**Comment by patriciarussellCAS** [comment]
I have reviewed in Test. Happy to have Cam do another test as well, but this is ready to go.
